public-ui/src/app/highlight-search/highlight-search.component.ts

```diff
@@ -23,4 +24,25 @@ export class HighlightSearchComponent implements OnChanges {
     }
     this.words = this.text.split(' ');
   }
+  public highlight(word) {
+    const splitWords = word.split(' ');
+    const temp = [];
+    for (let i = 0; i < splitWords.length; i++) {
```

Comment:
Took me a while to understand this, I'm not sure whether this would be clearer, but what do you think about something more like:

```
const matches = word.match(new RegExp(this.reString, 'g'));
const splits = word.split(new RegExp(this.reString));

// Interleave matches/splits.
for (let i = 0; i < matches.length; i++) {
  temp.push(splits[i], matches[i]);
}
temp.push(splits[splits.length-1]);
```

(also requires the `reString` is changed to be a string - current var name is a bit misleading)

Reply:
Fixed.